src/GitVersion.Core/Formatting/StringFormatWithExtension.cs

```diff
@@ -0,0 +1,100 @@
+using System.Text.RegularExpressions;
+using GitVersion.Core;
+
+namespace GitVersion.Formatting;
+
+internal static class StringFormatWithExtension
+{
+    internal static IExpressionCompiler ExpressionCompiler { get; set; } = new ExpressionCompiler();
+
+    internal static IInputSanitizer InputSanitizer { get; set; } = new InputSanitizer();
+
+    internal static IMemberResolver MemberResolver { get; set; } = new MemberResolver();
```

Comment:
Do we need the setters here?

Reply:
and also private instead of internal?

Reply:
and instead of property make them fields so that we do not create new instances of these types?

Reply:
Agreed.